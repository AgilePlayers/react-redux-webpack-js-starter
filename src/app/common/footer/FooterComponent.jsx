import React from 'react';

import './footer.scss';

const FooterComponent = () => (
  <div className="footer">
    <p className="footer-paragraph">Created by Kamil Bysiec from </p>
    <a
      href="https://agileplayers.com"
      target="_blank"
      rel="noopener noreferrer"
    >
      <img
        src="https://agileplayers.com/Static/dist/img/logo_full.svg"
        alt="agile players logo"
        className="footer-link-logo"
      />
    </a>
  </div>
);

export default FooterComponent;
