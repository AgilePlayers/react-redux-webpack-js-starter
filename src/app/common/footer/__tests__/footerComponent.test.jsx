import React from 'react';
import { shallow } from 'enzyme';
import FooterComponent from '../FooterComponent';

describe('common / footer / FooterComponent', () => {
  let component;

  beforeEach(() => {
    component = shallow(<FooterComponent />);
  });

  test('should render', () => {
    expect(component.length).toEqual(1);
  });

  test('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  test('should contains elements with "footer" and "footer-link" class', () => {
    expect(component.find('div.footer').length).toEqual(1);
    expect(component.find('a.footer-link').length).toEqual(1);
  });
});
