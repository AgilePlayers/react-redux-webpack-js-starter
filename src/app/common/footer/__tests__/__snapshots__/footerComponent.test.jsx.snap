// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common / footer / FooterComponent should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FooterComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="footer-paragraph"
        >
          Created by Kamil Bysiec from 
        </p>,
        <a
          href="https://agileplayers.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="agile players logo"
            className="footer-link-logo"
            src="https://agileplayers.com/Static/dist/img/logo_full.svg"
          />
        </a>,
      ],
      "className": "footer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Created by Kamil Bysiec from ",
          "className": "footer-paragraph",
        },
        "ref": null,
        "rendered": "Created by Kamil Bysiec from ",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="agile players logo"
            className="footer-link-logo"
            src="https://agileplayers.com/Static/dist/img/logo_full.svg"
          />,
          "href": "https://agileplayers.com",
          "rel": "noopener noreferrer",
          "target": "_blank",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "agile players logo",
            "className": "footer-link-logo",
            "src": "https://agileplayers.com/Static/dist/img/logo_full.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "a",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="footer-paragraph"
          >
            Created by Kamil Bysiec from 
          </p>,
          <a
            href="https://agileplayers.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="agile players logo"
              className="footer-link-logo"
              src="https://agileplayers.com/Static/dist/img/logo_full.svg"
            />
          </a>,
        ],
        "className": "footer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Created by Kamil Bysiec from ",
            "className": "footer-paragraph",
          },
          "ref": null,
          "rendered": "Created by Kamil Bysiec from ",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="agile players logo"
              className="footer-link-logo"
              src="https://agileplayers.com/Static/dist/img/logo_full.svg"
            />,
            "href": "https://agileplayers.com",
            "rel": "noopener noreferrer",
            "target": "_blank",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "agile players logo",
              "className": "footer-link-logo",
              "src": "https://agileplayers.com/Static/dist/img/logo_full.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "a",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
